File: frontend/src/constants/favorite.ts
export type FavoriteTargetType = 'plant' | 'article'

export const FavoriteTargetTypeMap: Record<FavoriteTargetType, string> = {
  plant: '植物品种',
  article: '养护文章',
}

export interface Favorite {
  id: number
  user_id: number
  target_type: FavoriteTargetType
  target_id: number
  created_at: string
}

File: backend/internal/model/favorite.go
package model

import "time"

// Favorite links a user to a plant or an article.
type Favorite struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	UserID     uint      `gorm:"index:idx_fav_user_target,unique;not null" json:"user_id"`
	TargetType string    `gorm:"size:16;index:idx_fav_user_target,unique;not null" json:"target_type"`
	TargetID   uint      `gorm:"index:idx_fav_user_target,unique;not null" json:"target_id"`
	CreatedAt  time.Time `json:"created_at"`
}
